### PR TITLE
SEOBundle : optimization of the extension NovaeESEOExtension

### PR DESCRIPTION
**Pull request by rhidja**
| Q             | A
| ------------- | ---
| Branch?       | master / x.y.z
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no

Hello @Plopix,

I would like to suggest for you this improvement of the the filter `computeMetas`, If we use `content` rather than `contentInfo`, can helps us to avoid many fetch data.

Ramtane.

### CODE REVIEW

**Comment by Plopix** [comment]
hello @rhidja can you fix the build? probably just coding standard! thank you for your contribution!

**Comment by rhidja** [comment]
@Plopix done!

File: components/SEOBundle/bundle/Core/MetaNameSchema.php [hunk 1]
```diff
@@ -112,19 +112,13 @@ public function setImageVariationService(VariationHandler $handler): void
         $this->imageVariationService = $handler;
     }
 
-    public function resolveMeta(Meta $meta, Content $content, ContentType $contentType = null): bool
```
Comment (Plopix) [inline components/SEOBundle/bundle/Core/MetaNameSchema.php:115]: Can we keep it an mark it deprecated? Like just a comment on top of the function. So we will remove in the next version

File: components/SEOBundle/bundle/Twig/NovaeZSEOExtension.php [hunk 2]
```diff
@@ -107,29 +106,20 @@ public function getFallbackedMetaContent(ContentInfo $contentInfo, string $metaN
     /**
      * Compute Metas of the Field thanks to its Content and the Fallback.
      */
-    public function computeMetas(Field $field, ContentInfo $contentInfo): string
+    public function computeMetas(Field $field, Content $content): string
```
Comment (Plopix) [inline components/SEOBundle/bundle/Twig/NovaeZSEOExtension.php:109]: this is BC break. We don't want to bump a version for that.
Can you remove the type-hint and test it.

If you receive `contentInfo` just convert to a content. And keep the rest.

Add a comment to mention that's to handle a deprecated code.

**Review by Plopix — CHANGES_REQUESTED** [review]
one more little change,

 `{{ field|compute_novaseometas(contentInfo) }}->{{ field|compute_novaseometas(content) }}`

This will break all installations after an update. So I put comments in the PR to make it work but not break (deprecation) and we will clean on the next major release.

Can you do that?

Thank you @rhidja !

File: components/SEOBundle/bundle/Twig/NovaeZSEOExtension.php (hunk 2 above)
Comment (rhidja) [reply components/SEOBundle/bundle/Twig/NovaeZSEOExtension.php]: @Plopix done!

File: components/SEOBundle/bundle/Core/MetaNameSchema.php (hunk 1 above)
Comment (rhidja) [reply components/SEOBundle/bundle/Core/MetaNameSchema.php:115]: @Plopix done!

**Comment by Plopix** [comment]
Nice @rhidja Merged!

**Comment by Plopix** [comment]
synced and tagged!

Thank you!!

**Pull request merged**